### PR TITLE
Deployment endpoints

### LINKED ISSUE #31: [Backend] Endpoint: GET /api/v1/deployments
**Übergeordnete User Story**
#46

**🎯 Ziel**
API-Endpunkt zur Auflistung aller Deployments des aktuellen Users.

**🧠 Beschreibung / Kontext**
Dozent/Student kann seine Deployments abrufen. Filter nach Status, Kurs, Template möglich. Pagination für große Listen.

**🔧 Aufgaben**
- [ ] Route `GET /api/v1/deployments` erstellen
- [ ] Query-Parameter: `status`, `course_id`, `template_id`, `page`, `page_size`
- [ ] Filter nach User-ID (aus Auth-Token)
- [ ] Response mit `ResponseBuilder.paginated()`

**✅ Akzeptanzkriterien**
- [ ] Anforderungen der User Story sind erfüllt
- [ ] Erfolgs- und Fehlerfälle sind korrekt behandelt
- [ ] Sinnvolle HTTP-Statuscodes
- [ ] Keine Breaking Changes

**🧪 Tests**
- [ ] Unit Tests
- [ ] Integration- / API-Tests
- [ ] Edge Cases abgedeckt

**📎 Abhängigkeiten**
- Blockiert durch: -
- Blockiert: -

### LINKED ISSUE #32: [Backend] Endpoint: GET /api/v1/deployments/{id}
**Übergeordnete User Story**
#46

**🎯 Ziel**
API-Endpunkt zur Abfrage eines einzelnen Deployments mit Details.

**🧠 Beschreibung / Kontext**
Dozent/Student kann Details eines Deployments abrufen: Status, Instances, Access-URLs, Logs.

**🔧 Aufgaben**
- [ ] Route `GET /api/v1/deployments/{id}` erstellen
- [ ] Berechtigungsprüfung (Owner oder Admin)
- [ ] Response mit vollständigen Deployment-Details inkl. Instances
- [ ] Fehlerbehandlung bei nicht gefundenem Deployment

**✅ Akzeptanzkriterien**
- [ ] Anforderungen der User Story sind erfüllt
- [ ] Erfolgs- und Fehlerfälle sind korrekt behandelt
- [ ] Sinnvolle HTTP-Statuscodes
- [ ] Keine Breaking Changes

**🧪 Tests**
- [ ] Unit Tests
- [ ] Integration- / API-Tests
- [ ] Edge Cases abgedeckt

**📎 Abhängigkeiten**
- Blockiert durch: -
- Blockiert: -

### LINKED ISSUE #36: [Backend] Minimal-Click Delete: DELETE /deployments/{id}
**Übergeordnete User Story**
#56

**🎯 Ziel**
Einfacher Endpunkt zum Löschen eines Deployments.

**🧠 Beschreibung / Kontext**
Admin/Dozent kann Deployment per DELETE-Request entfernen. Celery-Task löscht OpenStack-Ressourcen asynchron.

**🔧 Aufgaben**
- [ ] Route `DELETE /api/v1/deployments/{id}` implementieren
- [ ] Berechtigungsprüfung (Owner oder Admin)
- [ ] Celery-Task `delete_deployment.delay(deployment_id)` triggern
- [ ] Status des Deployments auf `deleting` setzen

**✅ Akzeptanzkriterien**
- [ ] Anforderungen der User Story sind erfüllt
- [ ] Erfolgs- und Fehlerfälle sind korrekt behandelt
- [ ] Sinnvolle HTTP-Statuscodes
- [ ] Keine Breaking Changes

**🧪 Tests**
- [ ] Unit Tests
- [ ] Integration- / API-Tests
- [ ] Edge Cases abgedeckt

**📎 Abhängigkeiten**
- Blockiert durch: -
- Blockiert: -

### LINKED ISSUE #50: [Backend] App-Restart: POST /api/v1/deployments/{id}/restart
**Übergeordnete User Story**
#52

**🎯 Ziel**
Endpunkt zum Neustart eines Deployments.

**🧠 Beschreibung / Kontext**
Dozent kann ein fehlerhaftes oder hängendes Deployment neu starten. Celery-Task führt Heat-Stack-Update oder VM-Reboot durch.

**🔧 Aufgaben**
- [ ] Route `POST /api/v1/deployments/{id}/restart` erstellen
- [ ] Celery-Task `restart_deployment.delay(deployment_id)` triggern
- [ ] Status-Update auf `restarting`
- [ ] Fehlerbehandlung bei laufenden Deployments

**✅ Akzeptanzkriterien**
- [ ] Anforderungen der User Story sind erfüllt
- [ ] Erfolgs- und Fehlerfälle sind korrekt behandelt
- [ ] Sinnvolle HTTP-Statuscodes
- [ ] Keine Breaking Changes

**🧪 Tests**
- [ ] Unit Tests
- [ ] Integration- / API-Tests
- [ ] Edge Cases abgedeckt

**📎 Abhängigkeiten**
- Blockiert durch: -
- Blockiert: -

### PR DESCRIPTION
**Pull request by nicowre**
## Linked Issue
Closes #31 #32 #36 #50 

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes)

## Checklist
- [X] My code follows the style guidelines of this project
- [X] I have added/updated unit tests
- [X] I have updated the documentation (if applicable)
- [X] I have verified my changes locally

## Screenshots / Logs (Optional)

### CODE REVIEW

File: src/core/config.py [hunk 1]
```diff
@@ -30,6 +30,16 @@ class Settings(BaseSettings):
     # Encryption key for secrets (Fernet key)
     encryption_key: str | None = None
     
+    # OpenStack fallback credentials (optional - used when no user credentials exist)
```
Comment (Dilmand) [inline src/core/config.py]: Die Openstack Credentials brauchen wir hier nicht

**Review by Dilmand — CHANGES_REQUESTED** [review]

**Review by Dilmand — APPROVED** [review]
